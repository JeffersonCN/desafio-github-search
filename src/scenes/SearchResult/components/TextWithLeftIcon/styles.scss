.text-with-left-icon {
    padding: 0 0 10px;

    .icon {
        width: 24px;
        height: 24px;
        margin-right: 7px;
        float: left;
    }

    .text {
        font-size: 20px;
        line-height: 24px;
        color: #5c5c5c;
    }
}