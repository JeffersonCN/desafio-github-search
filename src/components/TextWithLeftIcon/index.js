import React, { Component } from 'react';

import styles from './styles.scss';

class TextWithLeftIcon extends Component {
    render() {
        return (
            <div className={styles.textWithLeftIcon}>
                <img className={styles.icon} src={this.props.icon} alt={this.props.text} />
                <span className={styles.text}>{this.props.text}</span>
            </div>
        );
    }
}

export default TextWithLeftIcon;