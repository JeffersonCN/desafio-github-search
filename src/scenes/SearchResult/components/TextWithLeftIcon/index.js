import React, { Component } from 'react';

import './styles.scss';

class TextWithLeftIcon extends Component {
    render() {
        return (
            <div className="text-with-left-icon">
                <img className="icon" src={this.props.icon} alt={this.props.text} />
                <span className="text">{this.props.text}</span>
            </div>
        );
    }
}

export default TextWithLeftIcon;