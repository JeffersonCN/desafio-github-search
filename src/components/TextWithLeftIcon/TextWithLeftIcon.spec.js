import React from 'react';
import { shallow } from 'enzyme';
import TextWithLeftIcon from './';

describe('Testing TextWithLeftIcon Component', () => {
    it('shoud render correctly', () => {
        const wrapper = shallow(
            <TextWithLeftIcon />
        );

        expect(wrapper).toMatchSnapshot();
    });
});