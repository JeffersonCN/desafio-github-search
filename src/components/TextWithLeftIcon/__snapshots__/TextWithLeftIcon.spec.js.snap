// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TextWithLeftIcon Component shoud render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextWithLeftIcon />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt={undefined}
          className={undefined}
          src={undefined}
/>,
        <span
          className={undefined}
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "className": undefined,
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt={undefined}
            className={undefined}
            src={undefined}
/>,
          <span
            className={undefined}
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "className": undefined,
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
